### DISCUSSION TITLE
How to access system tables?

### DISCUSSION THREAD

**Discussion by ElierHG** [pocketbase/pocketbase#5864] [Q&A] [answered]
Does anyone know how to access system tables from the backend in JS routes, those that start with an underscore, like _admins?

**Comment by chnfyi** [comment]
In v0.22.24

```javascript
$app.dao().findAdminById(id)
$app.dao().findAdminByEmail(email)
``` 

in v0.23.0rc

```javascript
$app.findRecordById("_superusers", id)
$app.findAuthRecordByEmail("_superusers", email)
```

**Reply by ElierHG** [reply]
Thanks for the reply. 

I'm not trying to look up an admin per say, just wondering if there is a way of querying tables of that kind. I'm working on a project that has a tabled called **_pbl_apps** and I want to retrieve all rows, (this project [here](https://github.com/pedrozadotdev/pocketblocks)).

Sorry if the question seems too naive, I'm new to PB.

**Reply by chnfyi** [reply]
You can try:

```javascript
const records = arrayOf(new Record)

// v0.22.24
$app.dao().recordQuery("_pbl_apps").all(records)
    
// v0.23.0rc
$app.recordQuery("_pbl_apps").all(records)

// return records
``` 

If you need more methods to filter your query, you can take a look to: 
- https://pocketbase.io/docs/js-records/#custom-record-query
- https://pocketbase.io/docs/js-database/

**Reply by ElierHG** [reply]
I get the following error when doing:

`routerAdd("GET", '/mApi/apps', (c) => {

    const records = arrayOf(new Record)

    // v0.22.24
    $app.dao().recordQuery("_pbl_apps").all(records)
    
    console.log(records);

    c.json({ ok: true, data: records })
})`

![image](https://github.com/user-attachments/assets/dcb091d3-71f0-4a10-8dce-6c55a6c710d4)

I'm running version v0.22.14. 
Thanks again for your help.

**Reply by chnfyi** [reply]
Maybe you can try this instead `const records = $app.dao().findRecordsByExpr("_pbl_apps")`.

If you get the same error, please make sure that it's the correct collection name. Otherwise for now i have no idea of the issue.

**Reply by ElierHG** [reply]
Yeah, I get the same error, I'll continue exploring. 
Thanks

**Reply by chnfyi** [reply]
It could be a SQL-related issue with escaping of the underscore character.

**Reply by ganigeorgiev** [reply, marked as answer]
@chnfyi It is not. The `invalidCollectionModelOrIdentifier` is returned when an invalid or non-existing collection name is used.

@ElierHG If you are trying to query non-Record model data, then `recordQuery` is not the method you should use. See the raw select example in https://pocketbase.io/docs/js-database/#execute-all.

In any case, please try to elaborate more clearly what actually you are trying to do to avoid the back-and-forth.

**Reply by ElierHG** [reply]
@ganigeorgiev I just needed to query all records from a table called **_pbl_apps**, I just didn't know how. 

I followed what [https://pocketbase.io/docs/js-database/#execute-all](https://pocketbase.io/docs/js-database/#execute-all) says and got the results I needed. Thank you for your response.
